File: Leetcode 0628 Maximum Product of Three Numbers.go
// 2026/08/10

func maximumProduct(nums []int) int {
    sort.Ints(nums)
    n := len(nums)

    a := nums[n-1] * nums[n-2] * nums[n-3]
    b := nums[0] * nums[1] * nums[n-1]
    return max(a, b)
}

File: Leetcode 0628 Maximum Product of Three Numbers.js
// 2026/08/10
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumProduct = function(nums) {
    const n = nums.length
    nums.sort((a,b)=>a-b)
    return Math.max(nums[n-1]*nums[n-2]*nums[n-3], nums[n-1]*nums[0]*nums[1])
};
